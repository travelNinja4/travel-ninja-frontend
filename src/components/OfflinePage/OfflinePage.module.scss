/**
------------------------------------------------------
styles specific to the OfflinePage component
------------------------------------------------------
**/

.container {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  height: 100vh;
  text-align: center;
  padding: 1.5rem;
  background-color: var(--color-gray-25);
  animation: fadeIn 0.5s ease-in-out forwards;
}

.floating {
  font-size: 3.75rem;
  display: inline-block;
  animation: float 2.5s ease-in-out infinite;
}

.title {
  font-size: 1.875rem;
  font-weight: 700;
  margin-top: 1.5rem;
}

.message {
  color: var(--color-gray-320);
  margin-top: 0.75rem;
}

.waiting {
  margin-top: 2.5rem;
  font-size: 0.875rem;
  color: var(--color-gray-350);
  animation: pulse 2s infinite;
}

/* Fade-in animation for the container */
@keyframes fadeIn {
  0% {
    opacity: 0;
  }
  100% {
    opacity: 1;
  }
}

/* Pulse animation for waiting text */
@keyframes pulse {
  0%,
  100% {
    opacity: 1;
  }
  50% {
    opacity: 0.5;
  }
}
