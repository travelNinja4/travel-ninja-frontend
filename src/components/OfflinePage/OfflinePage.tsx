/**
 * A React component that shows a floating offline icon and message when the user is not connected to the internet.
 *
 * @example
 * ```tsx
 * import OfflinePage from '@src/components/OfflinePage'
 *
 * export default function OfflinePage() {
 *   return <OfflinePage label="Hello" />;
 * }
 * ```
 */

import { WifiOff } from 'lucide-react';
import Typography from '../Typography';
import { STRINGS } from '@/constants/strings';
import styles from './OfflinePage.module.scss';

export default function OfflinePage() {
  return (
    <div data-testid="OfflinePageTest" className={styles.container}>
      <WifiOff size={64} color="var(--color-red-500)" />
      <Typography tag="h1" className={styles.title}>
        {STRINGS.YOUR_OFFLINE}
      </Typography>
      <Typography tag="p" className={styles.message}>
        {STRINGS.PLEASE_CONNECT_TO_INTERNET}
      </Typography>
      <div className={styles.waiting}>{STRINGS.WAITING_FOR_CONNECTION}</div>
    </div>
  );
}
