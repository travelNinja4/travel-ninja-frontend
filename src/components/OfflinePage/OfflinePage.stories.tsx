import type { Meta, StoryObj } from '@storybook/nextjs-vite';
/**
import { fn } from 'storybook/test';
*/
import OfflinePage from './OfflinePage';

const meta = {
  title: 'components/OfflinePage',
  component: OfflinePage,
  parameters: {
    layout: 'centered',
  },
  tags: ['autodocs'],
  argTypes: {},
  args: {
    /** onClick: fn() **/
  },
} satisfies Meta<typeof OfflinePage>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = {
  args: {},
};
