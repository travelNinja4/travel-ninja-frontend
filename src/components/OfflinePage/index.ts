import OfflinePage from './OfflinePage';

export default OfflinePage;
