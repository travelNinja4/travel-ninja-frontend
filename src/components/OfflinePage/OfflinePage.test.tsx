import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import OfflinePage from './OfflinePage';

describe('Home', () => {
  /** Base props for OfflinePage **/
  const baseProps = {};

  /**
   * @test
   * Renders the OfflinePage component without throwing an error
   * and checks if the OfflinePageTest text is present in the document.
   *
   * This ensures that the component loads correctly with the base props.
   */
  it('renders without crashing', () => {
    render(<OfflinePage {...baseProps} />);
    expect(screen.getByTestId('OfflinePageTest')).toBeInTheDocument();
  });

  /**
   * @test
   * Ensures that the OfflinePage component renders without throwing any runtime errors.
   *
   * This is a safety check to confirm that the component can mount successfully
   * with the provided base props, even if it does not render any visible content.
   */
  it('renders without throwing', () => {
    expect(() => render(<OfflinePage {...baseProps} />)).not.toThrow();
  });

  /**
   * @test
   * Captures a snapshot of the OfflinePage component with base props
   * and ensures it matches the previously stored snapshot.
   *
   * This helps detect any unexpected UI changes.
   */
  it('matches snapshot', () => {
    const { asFragment } = render(<OfflinePage {...baseProps} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
