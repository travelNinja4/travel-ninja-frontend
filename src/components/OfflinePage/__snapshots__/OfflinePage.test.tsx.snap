// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Home matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
    data-testid="OfflinePageTest"
  >
    <svg
      aria-hidden="true"
      class="lucide lucide-wifi-off"
      fill="none"
      height="64"
      stroke="var(--color-red-500)"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="64"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M12 20h.01"
      />
      <path
        d="M8.5 16.429a5 5 0 0 1 7 0"
      />
      <path
        d="M5 12.859a10 10 0 0 1 5.17-2.69"
      />
      <path
        d="M19 12.859a10 10 0 0 0-2.007-1.523"
      />
      <path
        d="M2 8.82a15 15 0 0 1 4.177-2.643"
      />
      <path
        d="M22 8.82a15 15 0 0 0-11.288-3.764"
      />
      <path
        d="m2 2 20 20"
      />
    </svg>
    <h1
      class="typography h1 align-left title"
      data-testid="TypographyTest"
    >
      Youâ€™re offline!
    </h1>
    <p
      class="typography p align-left message"
      data-testid="TypographyTest"
    >
      Please reconnect to the internet to continue.
    </p>
    <div
      class="waiting"
    >
      Waiting for connection...
    </div>
  </div>
</DocumentFragment>
`;
